application/common/models/User.php

```diff
@@ -294,48 +295,105 @@ private function validatePassword(): Closure
         };
     }
 
-    /**
+    /*
+     * Check the following to decide if HIBP should be called:
+     *  - if Yii::$app->params['hibpCheckOnLogin'] is true
+     *  - if $this->password is not empty
+     *
+     */
+    public function shouldHibpBeChecked(): bool
+    {
+        return (\Yii::$app->params['hibpCheckOnLogin'] &&
+                ! empty($this->password) &&
+                time() >= strtotime($this->currentPassword->check_hibp_after) &&
+                $this->currentPassword->hibp_is_pwned == 'no');
+    }
```

Comment:
Excellent improvement! This seems much easier to read, including the change from ||'s to &&'s.